crates/exec-wasmtime/src/runtime/net/tls.rs

```diff
@@ -270,7 +278,11 @@ impl WasiFile for Listener {
             .context("could not create new TLS connection")
             .map(Connection::Server)?;
 
-        let mut stream = Stream { tcp, tls };
+        let mut stream = Stream {
+            tcp,
+            tls,
+            nonblocking: false,
```

Comment:
you can query the current state of blocking with libc::fcntl(fd, libc::F_GETFL...) and check for O_NONBLOCK.

Reply:
Yes, but we don't need to do that

Reply:
We're unsetting any flags below anyway (as suggested by `man 2 accept`):
> On Linux, the new socket returned by accept() does not inherit file status flags such as O_NONBLOCK and O_ASYNC from the listening socket.  This behavior differs from the canonical BSD sockets implementation.  Portable programs should not rely on inheritance or noninheritance of file status flags and always explicitly set all required flags on the socket returned from accept().